src/lib/mirage_block_ocaml.ml

```diff
@@ -360,16 +371,23 @@ let process_one t =
             return Response.Delete
           | Request.Flush h ->
             let t = Handle.find_or_quit h in
-            Block.flush t.Handle.base
-            >>= fun () ->
-            return Response.Flush
+            let open Lwt.Infix in
+            (* Block.write_error <> Qcow.write_error *)
+            begin Block.flush t.Handle.base
+              >>= function
+              | Error `Disconnected -> Lwt.return (Error `Disconnected)
+              | Error `Is_read_only -> Lwt.return (Error `Is_read_only)
```

Comment:
Same here, you can write this as:

```
| Error _ as e -> Lwt.return e
| Ok () -> return Response.Flush
```

and you can also probably remove the `begin ... end`